### ISSUE TITLE
Add new layer to the stack

### ISSUE THREAD

**Issue by ayushev** [hso-esk/emb6#22]
You will need to add new folder for [current repo structure](https://github.com/hso-esk/emb6/tree/develop/emb6/src) -> sock.
Paste sockets files there.
Add rules for scons script
There is also a need to add new layer into network stack structure in [emb6.h](https://github.com/hso-esk/emb6/blob/develop/emb6/emb6.h#L213) files

```
typedef struct netstack {
    const struct netstack_socket*                   sock;
    const struct netstack_headerCompression*        hc;
    const struct netstack_llsec*                    llsec;
    const struct netstack_highMac*                  hmac;
    const struct netstack_lowMac*                   lmac;
    const struct netstack_framer*                   frame;
    const struct netstack_interface*                inif;
    uint8_t                                         c_configured;
}s_ns_t;

typedef const struct netstack_socket{
    char *name;

    /** Initialize the network driver */
    void (* create)(s_ns_t* p_ns);
   /* Connect to remote node*/
    void (* connect)(...);
    void (* bind)(...);
    /* And so one and so forth */
}s_nsSocket_t;
```

New layer should have BSD-like interface
Have a look on how other interfaces are implemented

In other words application (be it in demo folder like udp-alive or coap in emb6/apl/coap) which desire to use tcp/udp should use sock->"function_name" calls

**Comment by edschmitt** [comment]
- the uip related function in the application layer are replaced with the udp-socket function
- at the moment there is NO automatic configurable socket interface (for UDP and TCPIP) available
- aptb is not tested yet
- coap is tested (udp-alive, ping, get temperture, get rf-info)
